packages/@glimmer-workspace/integration-tests/test/collections/object-test.ts

```diff
@@ -0,0 +1,207 @@
+import { trackedObject } from '@glimmer/validator';
+import {
+  defineComponent,
+  GlimmerishComponent as Component,
+  jitSuite,
+  RenderTest,
+  test,
+} from '@glimmer-workspace/integration-tests';
+
+class TrackedObjectTest extends RenderTest {
+  static suiteName = `trackedObject() (rendering)`;
+
+  @test
+  'it works when used directly in a template'() {
+    const obj = trackedObject<Record<PropertyKey, unknown>>({ foo: 123 });
+
+    const Foo = defineComponent({ obj }, '{{obj.foo}}');
+
+    this.renderComponent(Foo);
+
+    this.assertHTML('123');
+
+    obj['foo'] = 456;
+    this.rerender();
+
+    this.assertHTML('456');
+    this.assertStableRerender();
+  }
+
+  @test
+  'options.equals: default equals does not dirty on no-op changes'(assert: Assert) {
+    const obj = trackedObject({ foo: '123' });
+    const step = (x: string) => {
+      assert.step(x);
+      return x;
+    };
+
+    const Foo = defineComponent({ step, obj }, '{{step obj.foo}}');
+
+    this.renderComponent(Foo);
+
+    this.assertHTML('123');
+    assert.verifySteps(['123']);
+
+    obj['foo'] = '123';
+    this.rerender();
+
+    this.assertHTML('123');
+    this.assertStableRerender();
+    assert.verifySteps([]);
+  }
+
+  @test
+  'options.equals: using equals can dirty on every change'(assert: Assert) {
+    const obj = trackedObject({ foo: '123' }, { equals: () => false });
+    const step = (x: string) => {
+      assert.step(x);
+      return x;
+    };
+
+    const Foo = defineComponent({ step, obj }, '{{step obj.foo}}');
+
+    this.renderComponent(Foo);
+
+    this.assertHTML('123');
+    assert.verifySteps(['123']);
+
+    obj['foo'] = '123';
+    this.rerender();
+
+    this.assertHTML('123');
+    this.assertStableRerender();
+    assert.verifySteps(['123']);
+  }
+
+  @test
+  'each: set existing value'() {
+    this.assertEachReactivity(
+      class extends Component {
+        collection = trackedObject({ foo: 123 }, { equals: () => false });
+        update() {
+          this.collection.foo = 789;
```

Comment:
commenting this line out has no effect, looks like the test approach isn't correct?